### PR TITLE
Add methods for accessing sets near a value in RangeSet and RangeMap

### LINKED ISSUE #40: Add methods for getting the sets "closest" to a value in a RangeMap and RangeSet

### PR DESCRIPTION
**Pull request by rgabbard**
Adds maximal_containing_or_below and minimal_containing_or_above to
RangeSet. These allow querying for a value and getting the set
containing it, if any, or the next one in either direction.

Adds corresponding methods to RangeMap which work similarly,
but return the associated value.

Closes #40

### CODE REVIEW

**Comment by rgabbard** [comment]
Ready for review but WIP while I give it further testing in NLP code.

**Comment by rgabbard** [comment]
Would names like `get_at_or_above` and `get_at_or_below` be clearer for `RangeMap`?  Similarly `at_or_above` and `at_or_below` for `RangeSet`?

**Comment by ConstantineLignos** [comment]
I can review on Monday.

File: tests/test_range_set.py [hunk 1]
```diff
@@ -216,6 +216,136 @@ def test_all_two_ranges_intersecting(self):
             for query_2 in TestRangeSet.QUERY_RANGES:
                 self._test_intersects(RangeSet.create_mutable().add(query_1).add(query_2))
 
+    # forms the basis for corresponding tests in test_range_set
```
Comment (berquist) [inline tests/test_range_set.py]:
```suggestion
    # forms the basis for corresponding tests in test_range_map
```

**Comment by ConstantineLignos** [comment]
I don't have time to review in detail, but it looks good at a glance and I think @berquist's review is plenty.

**Pull request merged**